mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala

```diff
@@ -717,6 +719,49 @@ class SparseVector(
       new SparseVector(size, ii, vv)
     }
   }
+
+  override def argmax: Int = {
+    if (size == 0) {
+      -1
+    } else {
+
+      //grab first active index and value by default
+      var maxIdx = indices(0)
+      var maxValue = values(0)
+
+      foreachActive { (i, v) =>
+        if (v > maxValue) {
+          maxIdx = i
+          maxValue = v
+        }
+      }
+
+      // look for inactive values incase all active node values are negative
+      if(size != values.size && maxValue <= 0){
```

Comment:
I had meant something like this.

    val a = new SparseVector(3, Array(0, 1), Array(0, -1))
    
Till this block of code, the maxIdx would be 0 and maxValue would be 0 and since the condition `size != values.size && maxValue <= 0` satisfies, it would return the first inactive Index, i.e 2. However we want 0 to be returned.